spec/views/post_spec.rb

```diff
@@ -0,0 +1,133 @@
+require 'rails_helper'
+
+RSpec.describe 'Post', type: :feature do
+  before :all do
+    @first_user = User.find_by(name: 'Tom')
+    if @first_user.nil?
+      @first_user = User.create(name: 'Tom', photo: 'https://placeholder.com', bio: 'User\'s bio', password: '222555',
+                                email: 'tom@example.com')
+    end
+    visit new_user_session_path
+    fill_in 'Email', with: 'tom@example.com'
+    fill_in 'Password', with: '222555'
+    click_button 'Log in'
+
+    @post = @first_user.posts.find_by(title: 'Post title 1')
+    @post = @first_user.posts.create!(title: 'Post title 1', text: 'Post text 1') if @post.nil?
+    if @post.comments.empty?
+      @comment1 = @post.comments.create(text: 'Comment text 1', user_id: @first_user.id)
+      @comment2 = @post.comments.create(text: 'Comment text 2', user_id: @first_user.id)
+      @comment3 = @post.comments.create(text: 'Comment text 3', user_id: @first_user.id)
+      @comment4 = @post.comments.create(text: 'Comment text 4', user_id: @first_user.id)
+      @comment5 = @post.comments.create(text: 'Comment text 5', user_id: @first_user.id)
+    else
+      @comment1 = @post.comments.find(1)
+      @comment2 = @post.comments.find(2)
+      @comment3 = @post.comments.find(3)
+      @comment4 = @post.comments.find(4)
+      @comment5 = @post.comments.find(5)
+    end
+  end
+
+  describe 'index' do
+    before :each do
+      visit user_posts_path(@first_user)
+    end
+
+    it 'See the profile picture for each user' do
+      all_images = page.all('img')
+      expect(all_images.count).to eq(1)
+    end
+
+    it 'See the user\'s username' do
+      expect(page).to have_content('Tom')
+    end
+
+    it 'See the number of posts each user has written' do
+      expect(page).to have_content("Posts: #{@first_user.posts.count}")
+    end
+
+    it 'See the post title' do
+      expect(page).to have_content(@post.title)
+    end
```

Comment:
- [ ] It would be better if we are able to pass all our test cases. Let us kindly fix the syntax error from our post controller
![Screenshot from 2022-04-06 23-37-26](https://user-images.githubusercontent.com/46388113/162091778-c68aab12-7d70-4a20-b3ad-0433734375bf.png)